backend/fms_core/viewsets/sample.py

```diff
@@ -23,7 +23,13 @@
 
 
 class SampleViewSet(viewsets.ModelViewSet, TemplateActionsMixin, TemplatePrefillsMixin):
-    queryset = Sample.objects.select_related("container").all().distinct()
+    queryset = Sample.objects.annotate(
```

Comment:
apply the annotation on the previous query instead of inserting it in the chain. it would be more readable to have them separated (example
```
queryset = sample.objects.filter(name=bob).all()
queryset = queryset.annotate(patate_poil=Q()& Q()))
```